validator/validator.go

```diff
@@ -0,0 +1,76 @@
+package validator
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"gopkg.in/Clever/kayvee-go.v2/logger"
+)
+
+var (
+	requiredKayveeFields = []string{
+		"source",
+		"title",
+		"level",
+	}
+
+	validLevels = map[string]bool{
+		logger.Debug.String():    true,
+		logger.Info.String():     true,
+		logger.Warning.String():  true,
+		logger.Error.String():    true,
+		logger.Critical.String(): true,
+	}
+)
+
+// ValidateFormat returns an error if the given string is not a valid
+// JSON-formatted kayvee log line.
+func ValidateJSONFormat(logLine string) error {
+	var kayveeData map[string]interface{}
+
+	err := json.Unmarshal([]byte(logLine), &kayveeData)
+	if err != nil {
+		return err
+	}
+
+	err = checkRequiredFields(kayveeData, requiredKayveeFields)
+	if err != nil {
+		return err
+	}
+
+	// Check for type-specific required fields.
+	if kayveeData["type"] != nil {
+		logType, isString := kayveeData["type"].(string)
+		if !isString {
+			return fmt.Errorf("Invalid log type `%s`", kayveeData["type"])
+		}
+
+		var typeRequiredFields []string
+		switch logType {
+		case "gauge":
+			typeRequiredFields = []string{"value"}
+		}
+
+		err = checkRequiredFields(kayveeData, typeRequiredFields)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Check for a valid `level` field.
+	if level, isString := kayveeData["level"].(string); !isString || !validLevels[level] {
+		return fmt.Errorf("Invalid log level `%s`", kayveeData["level"])
+	}
+
+	return nil
+}
+
+func checkRequiredFields(kayveeData map[string]interface{}, fields []string) error {
+	for _, field := range fields {
+		if kayveeData[field] == nil {
+			return fmt.Errorf("Missing required field `%s`", field)
```

Comment:
I'd prefer returning _all_ missing required fields, not just the first one detected. That said, if you're missing 2 or 3 out of just 3 required fields, :angry: